{
 "matches": [
  {
   "vulnerability": {
    "id": "ALAS-2024-2607",
    "dataSource": "https://alas.aws.amazon.com/AL2/ALAS-2024-2607.html",
    "namespace": "amazon:distro:amazonlinux:2",
    "severity": "Low",
    "urls": [
     "https://alas.aws.amazon.com/AL2/ALAS-2024-2607.html"
    ],
    "description": "Certifi is a curated collection of Root Certificates for validating the trustworthiness of SSL certificates while verifying the identity of TLS hosts. Certifi starting in 2021.05.30 and prior to 2024.07.4 recognized root certificates from `GLOBALTRUST`. Certifi 2024.07.04 removes root certificates from `GLOBALTRUST` from the root store. These are in the process of being removed from Mozilla's trust store. `GLOBALTRUST`'s root certificates are being removed pursuant to an investigation which identified \"long-running and unresolved compliance issues.\" (CVE-2024-39689)",
    "cvss": [],
    "fix": {
     "versions": [
      "2023.2.68-1.amzn2.0.1"
     ],
     "state": "fixed"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [
    {
     "id": "CVE-2024-39689",
     "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-39689",
     "namespace": "nvd:cpe",
     "severity": "High",
     "urls": [
      "https://github.com/certifi/python-certifi/commit/bd8153872e9c6fc98f4023df9c2deaffea2fa463",
      "https://github.com/certifi/python-certifi/security/advisories/GHSA-248v-346w-9cwc",
      "https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/XpknYMPO8dI"
     ],
     "description": "Certifi is a curated collection of Root Certificates for validating the trustworthiness of SSL certificates while verifying the identity of TLS hosts. Certifi starting in 2021.05.30 and prior to 2024.07.4 recognized root certificates from `GLOBALTRUST`. Certifi 2024.07.04 removes root certificates from `GLOBALTRUST` from the root store. These are in the process of being removed from Mozilla's trust store. `GLOBALTRUST`'s root certificates are being removed pursuant to an investigation which identified \"long-running and unresolved compliance issues.\"",
     "cvss": [
      {
       "source": "security-advisories@github.com",
       "type": "Secondary",
       "version": "3.1",
       "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N",
       "metrics": {
        "baseScore": 7.5,
        "exploitabilityScore": 3.9,
        "impactScore": 3.6
       },
       "vendorMetadata": {}
      }
     ]
    }
   ],
   "matchDetails": [
    {
     "type": "exact-direct-match",
     "matcher": "rpm-matcher",
     "searchedBy": {
      "distro": {
       "type": "amazonlinux",
       "version": "2"
      },
      "namespace": "amazon:distro:amazonlinux:2",
      "package": {
       "name": "ca-certificates",
       "version": "0:2023.2.64-1.amzn2.0.1"
      }
     },
     "found": {
      "versionConstraint": "< 2023.2.68-1.amzn2.0.1 (rpm)",
      "vulnerabilityID": "ALAS-2024-2607"
     }
    }
   ],
   "artifact": {
    "id": "3fdee4f2791be89b",
    "name": "ca-certificates",
    "version": "2023.2.64-1.amzn2.0.1",
    "type": "rpm",
    "locations": [
     {
      "path": "/var/lib/rpm/Packages",
      "layerID": "sha256:0044b36762beca646f20eb6730e3575ec671a8857b39485f38ba0e715d375483"
     }
    ],
    "language": "",
    "licenses": [
     "Public Domain"
    ],
    "cpes": [
     "cpe:2.3:a:ca-certificates:ca-certificates:2023.2.64-1.amzn2.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:ca-certificates:ca_certificates:2023.2.64-1.amzn2.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:ca_certificates:ca-certificates:2023.2.64-1.amzn2.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:ca_certificates:ca_certificates:2023.2.64-1.amzn2.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:amazonlinux:ca-certificates:2023.2.64-1.amzn2.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:amazonlinux:ca_certificates:2023.2.64-1.amzn2.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:ca:ca-certificates:2023.2.64-1.amzn2.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:ca:ca_certificates:2023.2.64-1.amzn2.0.1:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:rpm/amzn/ca-certificates@2023.2.64-1.amzn2.0.1?arch=noarch&upstream=ca-certificates-2023.2.64-1.amzn2.0.1.src.rpm&distro=amzn-2",
    "upstreams": [],
    "metadataType": "RpmMetadata",
    "metadata": {
     "epoch": null,
     "modularityLabel": ""
    }
   }
  },
  {
   "vulnerability": {
    "id": "ALAS-2024-2595",
    "dataSource": "https://alas.aws.amazon.com/AL2/ALAS-2024-2595.html",
    "namespace": "amazon:distro:amazonlinux:2",
    "severity": "Medium",
    "urls": [
     "https://alas.aws.amazon.com/AL2/ALAS-2024-2595.html"
    ],
    "description": "krb5: GSS message token handling (CVE-2024-37370)In MIT Kerberos 5 (aka krb5) before 1.21.3, an attacker can cause invalid memory reads during GSS message token handling by sending message tokens with invalid length fields. (CVE-2024-37371)",
    "cvss": [],
    "fix": {
     "versions": [
      "1.15.1-55.amzn2.2.8"
     ],
     "state": "fixed"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [
    {
     "id": "CVE-2024-37370",
     "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-37370",
     "namespace": "nvd:cpe",
     "severity": "High",
     "urls": [
      "https://github.com/krb5/krb5/commit/55fbf435edbe2e92dd8101669b1ce7144bc96fef",
      "https://web.mit.edu/kerberos/www/advisories/"
     ],
     "description": "In MIT Kerberos 5 (aka krb5) before 1.21.3, an attacker can modify the plaintext Extra Count field of a confidential GSS krb5 wrap token, causing the unwrapped token to appear truncated to the application.",
     "cvss": [
      {
       "source": "nvd@nist.gov",
       "type": "Primary",
       "version": "3.1",
       "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N",
       "metrics": {
        "baseScore": 7.5,
        "exploitabilityScore": 3.9,
        "impactScore": 3.6
       },
       "vendorMetadata": {}
      }
     ]
    },
    {
     "id": "CVE-2024-37371",
     "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-37371",
     "namespace": "nvd:cpe",
     "severity": "Critical",
     "urls": [
      "https://github.com/krb5/krb5/commit/55fbf435edbe2e92dd8101669b1ce7144bc96fef",
      "https://web.mit.edu/kerberos/www/advisories/"
     ],
     "description": "In MIT Kerberos 5 (aka krb5) before 1.21.3, an attacker can cause invalid memory reads during GSS message token handling by sending message tokens with invalid length fields.",
     "cvss": [
      {
       "source": "nvd@nist.gov",
       "type": "Primary",
       "version": "3.1",
       "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:H",
       "metrics": {
        "baseScore": 9.1,
        "exploitabilityScore": 3.9,
        "impactScore": 5.2
       },
       "vendorMetadata": {}
      }
     ]
    }
   ],
   "matchDetails": [
    {
     "type": "exact-direct-match",
     "matcher": "rpm-matcher",
     "searchedBy": {
      "distro": {
       "type": "amazonlinux",
       "version": "2"
      },
      "namespace": "amazon:distro:amazonlinux:2",
      "package": {
       "name": "krb5-libs",
       "version": "0:1.15.1-55.amzn2.2.7"
      }
     },
     "found": {
      "versionConstraint": "< 1.15.1-55.amzn2.2.8 (rpm)",
      "vulnerabilityID": "ALAS-2024-2595"
     }
    },
    {
     "type": "exact-indirect-match",
     "matcher": "rpm-matcher",
     "searchedBy": {
      "distro": {
       "type": "amazonlinux",
       "version": "2"
      },
      "namespace": "amazon:distro:amazonlinux:2",
      "package": {
       "name": "krb5",
       "version": "1.15.1-55.amzn2.2.7"
      }
     },
     "found": {
      "versionConstraint": "< 1.15.1-55.amzn2.2.8 (rpm)",
      "vulnerabilityID": "ALAS-2024-2595"
     }
    }
   ],
   "artifact": {
    "id": "3bb967025c5652e0",
    "name": "krb5-libs",
    "version": "1.15.1-55.amzn2.2.7",
    "type": "rpm",
    "locations": [
     {
      "path": "/var/lib/rpm/Packages",
      "layerID": "sha256:0044b36762beca646f20eb6730e3575ec671a8857b39485f38ba0e715d375483"
     }
    ],
    "language": "",
    "licenses": [
     "MIT"
    ],
    "cpes": [
     "cpe:2.3:a:amazonlinux:krb5-libs:1.15.1-55.amzn2.2.7:*:*:*:*:*:*:*",
     "cpe:2.3:a:amazonlinux:krb5_libs:1.15.1-55.amzn2.2.7:*:*:*:*:*:*:*",
     "cpe:2.3:a:krb5-libs:krb5-libs:1.15.1-55.amzn2.2.7:*:*:*:*:*:*:*",
     "cpe:2.3:a:krb5-libs:krb5_libs:1.15.1-55.amzn2.2.7:*:*:*:*:*:*:*",
     "cpe:2.3:a:krb5_libs:krb5-libs:1.15.1-55.amzn2.2.7:*:*:*:*:*:*:*",
     "cpe:2.3:a:krb5_libs:krb5_libs:1.15.1-55.amzn2.2.7:*:*:*:*:*:*:*",
     "cpe:2.3:a:krb5:krb5-libs:1.15.1-55.amzn2.2.7:*:*:*:*:*:*:*",
     "cpe:2.3:a:krb5:krb5_libs:1.15.1-55.amzn2.2.7:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:rpm/amzn/krb5-libs@1.15.1-55.amzn2.2.7?arch=aarch64&upstream=krb5-1.15.1-55.amzn2.2.7.src.rpm&distro=amzn-2",
    "upstreams": [
     {
      "name": "krb5",
      "version": "1.15.1-55.amzn2.2.7"
     }
    ],
    "metadataType": "RpmMetadata",
    "metadata": {
     "epoch": null,
     "modularityLabel": ""
    }
   }
  },
  {
   "vulnerability": {
    "id": "ALAS-2024-2604",
    "dataSource": "https://alas.aws.amazon.com/AL2/ALAS-2024-2604.html",
    "namespace": "amazon:distro:amazonlinux:2",
    "severity": "Medium",
    "urls": [
     "https://alas.aws.amazon.com/AL2/ALAS-2024-2604.html"
    ],
    "description": "Issue summary: Calling the OpenSSL API function SSL_select_next_proto with anempty supported client protocols buffer may cause a crash or memory contents tobe sent to the peer.Impact summary: A buffer overread can have a range of potential consequencessuch as unexpected application beahviour or a crash. In particular this issuecould result in up to 255 bytes of arbitrary private data from memory being sentto the peer leading to a loss of confidentiality. However, only applicationsthat directly call the SSL_select_next_proto function with a 0 length list ofsupported client protocols are affected by this issue. This would normally neverbe a valid scenario and is typically not under attacker control but may occur byaccident in the case of a configuration or programming error in the callingapplication.The OpenSSL API function SSL_select_next_proto is typically used by TLSapplications that support ALPN (Application Layer Protocol Negotiation) or NPN(Next Protocol Negotiation). NPN is older, was never standardised andis deprecated in favour of ALPN. We believe that ALPN is significantly morewidely deployed than NPN. The SSL_select_next_proto function accepts a list ofprotocols from the server and a list of protocols from the client and returnsthe first protocol that appears in the server list that also appears in theclient list. In the case of no overlap between the two lists it returns thefirst item in the client list. In either case it will signal whether an overlapbetween the two lists was found. In the case where SSL_select_next_proto iscalled with a zero length client list it fails to notice this condition andreturns the memory immediately following the client list pointer (and reportsthat there was no overlap in the lists).This function is typically called from a server side application callback forALPN or a client side application callback for NPN. In the case of ALPN the listof protocols supplied by the client is guaranteed by libssl to never be zero inlength. The list of server protocols comes from the application and should nevernormally be expected to be of zero length. In this case if theSSL_select_next_proto function has been called as expected (with the listsupplied by the client passed in the client/client_len parameters), then theapplication will not be vulnerable to this issue. If the application hasaccidentally been configured with a zero length server list, and hasaccidentally passed that zero length server list in the client/client_lenparameters, and has additionally failed to correctly handle a \"no overlap\"response (which would normally result in a handshake failure in ALPN) then itwill be vulnerable to this problem.In the case of NPN, the protocol permits the client to opportunistically selecta protocol when there is no overlap. OpenSSL returns the first client protocolin the no overlap case in support of this. The list of client protocols comesfrom the application and should never normally be expected to be of zero length.However if the SSL_select_next_proto function is accidentally called with aclient_len of 0 then an invalid memory pointer will be returned instead. If theapplication uses this output as the opportunistic protocol then the loss ofconfidentiality will occur.This issue has been assessed as Low severity because applications are mostlikely to be vulnerable if they are using NPN instead of ALPN - but NPN is notwidely used. It also requires an application configuration or programming error.Finally, this issue would not typically be under attacker control making activeexploitation unlikely.The FIPS modules in 3.3, 3.2, 3.1 and 3.0 are not affected by this issue.Due to the low severity of this issue we are not issuing new releases ofOpenSSL at this time. The fix will be included in the next releases when theybecome available. (CVE-2024-5535)",
    "cvss": [],
    "fix": {
     "versions": [
      "1.0.2k-24.amzn2.0.13"
     ],
     "state": "fixed"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [
    {
     "id": "CVE-2024-5535",
     "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-5535",
     "namespace": "nvd:cpe",
     "severity": "Critical",
     "urls": [
      "http://www.openwall.com/lists/oss-security/2024/06/27/1",
      "http://www.openwall.com/lists/oss-security/2024/06/28/4",
      "https://github.com/openssl/openssl/commit/4ada436a1946cbb24db5ab4ca082b69c1bc10f37",
      "https://github.com/openssl/openssl/commit/99fb785a5f85315b95288921a321a935ea29a51e",
      "https://github.com/openssl/openssl/commit/cf6f91f6121f4db167405db2f0de410a456f260c",
      "https://github.com/openssl/openssl/commit/e86ac436f0bd54d4517745483e2315650fae7b2c",
      "https://github.openssl.org/openssl/extended-releases/commit/9947251413065a05189a63c9b7a6c1d4e224c21c",
      "https://github.openssl.org/openssl/extended-releases/commit/b78ec0824da857223486660177d3b1f255c65d87",
      "https://security.netapp.com/advisory/ntap-20240712-0005/",
      "https://www.openssl.org/news/secadv/20240627.txt"
     ],
     "description": "Issue summary: Calling the OpenSSL API function SSL_select_next_proto with an\nempty supported client protocols buffer may cause a crash or memory contents to\nbe sent to the peer.\n\nImpact summary: A buffer overread can have a range of potential consequences\nsuch as unexpected application beahviour or a crash. In particular this issue\ncould result in up to 255 bytes of arbitrary private data from memory being sent\nto the peer leading to a loss of confidentiality. However, only applications\nthat directly call the SSL_select_next_proto function with a 0 length list of\nsupported client protocols are affected by this issue. This would normally never\nbe a valid scenario and is typically not under attacker control but may occur by\naccident in the case of a configuration or programming error in the calling\napplication.\n\nThe OpenSSL API function SSL_select_next_proto is typically used by TLS\napplications that support ALPN (Application Layer Protocol Negotiation) or NPN\n(Next Protocol Negotiation). NPN is older, was never standardised and\nis deprecated in favour of ALPN. We believe that ALPN is significantly more\nwidely deployed than NPN. The SSL_select_next_proto function accepts a list of\nprotocols from the server and a list of protocols from the client and returns\nthe first protocol that appears in the server list that also appears in the\nclient list. In the case of no overlap between the two lists it returns the\nfirst item in the client list. In either case it will signal whether an overlap\nbetween the two lists was found. In the case where SSL_select_next_proto is\ncalled with a zero length client list it fails to notice this condition and\nreturns the memory immediately following the client list pointer (and reports\nthat there was no overlap in the lists).\n\nThis function is typically called from a server side application callback for\nALPN or a client side application callback for NPN. In the case of ALPN the list\nof protocols supplied by the client is guaranteed by libssl to never be zero in\nlength. The list of server protocols comes from the application and should never\nnormally be expected to be of zero length. In this case if the\nSSL_select_next_proto function has been called as expected (with the list\nsupplied by the client passed in the client/client_len parameters), then the\napplication will not be vulnerable to this issue. If the application has\naccidentally been configured with a zero length server list, and has\naccidentally passed that zero length server list in the client/client_len\nparameters, and has additionally failed to correctly handle a \"no overlap\"\nresponse (which would normally result in a handshake failure in ALPN) then it\nwill be vulnerable to this problem.\n\nIn the case of NPN, the protocol permits the client to opportunistically select\na protocol when there is no overlap. OpenSSL returns the first client protocol\nin the no overlap case in support of this. The list of client protocols comes\nfrom the application and should never normally be expected to be of zero length.\nHowever if the SSL_select_next_proto function is accidentally called with a\nclient_len of 0 then an invalid memory pointer will be returned instead. If the\napplication uses this output as the opportunistic protocol then the loss of\nconfidentiality will occur.\n\nThis issue has been assessed as Low severity because applications are most\nlikely to be vulnerable if they are using NPN instead of ALPN - but NPN is not\nwidely used. It also requires an application configuration or programming error.\nFinally, this issue would not typically be under attacker control making active\nexploitation unlikely.\n\nThe FIPS modules in 3.3, 3.2, 3.1 and 3.0 are not affected by this issue.\n\nDue to the low severity of this issue we are not issuing new releases of\nOpenSSL at this time. The fix will be included in the next releases when they\nbecome available.",
     "cvss": [
      {
       "source": "134c704f-9b21-4f2e-91b3-4a467353bcc0",
       "type": "Secondary",
       "version": "3.1",
       "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:H",
       "metrics": {
        "baseScore": 9.1,
        "exploitabilityScore": 3.9,
        "impactScore": 5.2
       },
       "vendorMetadata": {}
      }
     ]
    }
   ],
   "matchDetails": [
    {
     "type": "exact-direct-match",
     "matcher": "rpm-matcher",
     "searchedBy": {
      "distro": {
       "type": "amazonlinux",
       "version": "2"
      },
      "namespace": "amazon:distro:amazonlinux:2",
      "package": {
       "name": "openssl-libs",
       "version": "1:1.0.2k-24.amzn2.0.12"
      }
     },
     "found": {
      "versionConstraint": "< 1.0.2k-24.amzn2.0.13 (rpm)",
      "vulnerabilityID": "ALAS-2024-2604"
     }
    },
    {
     "type": "exact-indirect-match",
     "matcher": "rpm-matcher",
     "searchedBy": {
      "distro": {
       "type": "amazonlinux",
       "version": "2"
      },
      "namespace": "amazon:distro:amazonlinux:2",
      "package": {
       "name": "openssl",
       "version": "1.0.2k-24.amzn2.0.12"
      }
     },
     "found": {
      "versionConstraint": "< 1.0.2k-24.amzn2.0.13 (rpm)",
      "vulnerabilityID": "ALAS-2024-2604"
     }
    }
   ],
   "artifact": {
    "id": "ca07aa6bdda5c274",
    "name": "openssl-libs",
    "version": "1:1.0.2k-24.amzn2.0.12",
    "type": "rpm",
    "locations": [
     {
      "path": "/var/lib/rpm/Packages",
      "layerID": "sha256:0044b36762beca646f20eb6730e3575ec671a8857b39485f38ba0e715d375483"
     }
    ],
    "language": "",
    "licenses": [
     "OpenSSL"
    ],
    "cpes": [
     "cpe:2.3:a:openssl-libs:openssl-libs:1:1.0.2k-24.amzn2.0.12:*:*:*:*:*:*:*",
     "cpe:2.3:a:openssl-libs:openssl_libs:1:1.0.2k-24.amzn2.0.12:*:*:*:*:*:*:*",
     "cpe:2.3:a:openssl_libs:openssl-libs:1:1.0.2k-24.amzn2.0.12:*:*:*:*:*:*:*",
     "cpe:2.3:a:openssl_libs:openssl_libs:1:1.0.2k-24.amzn2.0.12:*:*:*:*:*:*:*",
     "cpe:2.3:a:amazonlinux:openssl-libs:1:1.0.2k-24.amzn2.0.12:*:*:*:*:*:*:*",
     "cpe:2.3:a:amazonlinux:openssl_libs:1:1.0.2k-24.amzn2.0.12:*:*:*:*:*:*:*",
     "cpe:2.3:a:openssl:openssl-libs:1:1.0.2k-24.amzn2.0.12:*:*:*:*:*:*:*",
     "cpe:2.3:a:openssl:openssl_libs:1:1.0.2k-24.amzn2.0.12:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:rpm/amzn/openssl-libs@1.0.2k-24.amzn2.0.12?arch=aarch64&epoch=1&upstream=openssl-1.0.2k-24.amzn2.0.12.src.rpm&distro=amzn-2",
    "upstreams": [
     {
      "name": "openssl",
      "version": "1.0.2k-24.amzn2.0.12"
     }
    ],
    "metadataType": "RpmMetadata",
    "metadata": {
     "epoch": 1,
     "modularityLabel": ""
    }
   }
  },
  {
   "vulnerability": {
    "id": "GHSA-mq26-g339-26xf",
    "dataSource": "https://github.com/advisories/GHSA-mq26-g339-26xf",
    "namespace": "github:language:python",
    "severity": "Medium",
    "urls": [
     "https://github.com/advisories/GHSA-mq26-g339-26xf"
    ],
    "description": "Command Injection in pip when used with Mercurial",
    "cvss": [
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N",
      "metrics": {
       "baseScore": 5.5,
       "exploitabilityScore": 1.8,
       "impactScore": 3.6
      },
      "vendorMetadata": {
       "base_severity": "Medium",
       "status": "N/A"
      }
     }
    ],
    "fix": {
     "versions": [
      "23.3"
     ],
     "state": "fixed"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [
    {
     "id": "CVE-2023-5752",
     "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2023-5752",
     "namespace": "nvd:cpe",
     "severity": "Low",
     "urls": [
      "https://github.com/pypa/pip/pull/12306",
      "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/",
      "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/",
      "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/",
      "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/",
      "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/",
      "https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/"
     ],
     "description": "When installing a package from a Mercurial VCS URL  (ie \"pip install \nhg+...\") with pip prior to v23.3, the specified Mercurial revision could\n be used to inject arbitrary configuration options to the \"hg clone\" \ncall (ie \"--config\"). Controlling the Mercurial configuration can modify\n how and which repository is installed. This vulnerability does not \naffect users who aren't installing from Mercurial.\n",
     "cvss": [
      {
       "source": "nvd@nist.gov",
       "type": "Primary",
       "version": "3.1",
       "vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N",
       "metrics": {
        "baseScore": 3.3,
        "exploitabilityScore": 1.8,
        "impactScore": 1.4
       },
       "vendorMetadata": {}
      },
      {
       "source": "cna@python.org",
       "type": "Secondary",
       "version": "3.1",
       "vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N",
       "metrics": {
        "baseScore": 5.5,
        "exploitabilityScore": 1.8,
        "impactScore": 3.6
       },
       "vendorMetadata": {}
      }
     ]
    }
   ],
   "matchDetails": [
    {
     "type": "exact-direct-match",
     "matcher": "python-matcher",
     "searchedBy": {
      "language": "python",
      "namespace": "github:language:python",
      "package": {
       "name": "pip",
       "version": "23.0.1"
      }
     },
     "found": {
      "versionConstraint": "<23.3 (python)",
      "vulnerabilityID": "GHSA-mq26-g339-26xf"
     }
    }
   ],
   "artifact": {
    "id": "8cac57bcd656efc6",
    "name": "pip",
    "version": "23.0.1",
    "type": "python",
    "locations": [
     {
      "path": "/var/lang/lib/python3.9/site-packages/pip-23.0.1.dist-info/METADATA",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/python3.9/site-packages/pip-23.0.1.dist-info/RECORD",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/python3.9/site-packages/pip-23.0.1.dist-info/top_level.txt",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "python",
    "licenses": [
     "MIT"
    ],
    "cpes": [
     "cpe:2.3:a:pip_developers_project:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip_developers_project:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils_sig_project:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils_sig_project:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip_developersproject:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip_developersproject:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils_sigproject:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils_sigproject:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip_developers_project:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils_sig_project:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip_developers:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip_developers:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip_developersproject:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils-sig:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils-sig:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils_sig:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils_sig:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils_sigproject:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:python-pip:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:python-pip:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:python_pip:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:python_pip:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip_developers:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils-sig:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:distutils_sig:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pypa:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pypa:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip:python-pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip:python_pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:python-pip:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:python_pip:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pypa:pip:23.0.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:pip:pip:23.0.1:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:pypi/pip@23.0.1",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2024-7592",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-7592",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://github.com/python/cpython/issues/123067",
     "https://github.com/python/cpython/pull/123075",
     "https://mail.python.org/archives/list/security-announce@python.org/thread/HXJAAAALNUNGCQUS2W7WR6GFIZIHFOOK/"
    ],
    "description": "There is a LOW severity vulnerability affecting CPython, specifically the\n'http.cookies' standard library module.\n\n\nWhen parsing cookies that contained backslashes for quoted characters in\nthe cookie value, the parser would use an algorithm with quadratic\ncomplexity, resulting in excess CPU resources being used while parsing the\nvalue.",
    "cvss": [
     {
      "source": "nvd@nist.gov",
      "type": "Primary",
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
      "metrics": {
       "baseScore": 7.5,
       "exploitabilityScore": 3.9,
       "impactScore": 3.6
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
      ],
      "package": {
       "name": "python",
       "version": "3.9.19"
      }
     },
     "found": {
      "vulnerabilityID": "CVE-2024-7592",
      "versionConstraint": "<= 3.13.0 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "id": "a120648b38c61b3b",
    "name": "python",
    "version": "3.9.19",
    "type": "binary",
    "locations": [
     {
      "path": "/var/lang/bin/python3.9",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/libpython3.9.so.1.0",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.9.19",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2024-0397",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-0397",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "http://www.openwall.com/lists/oss-security/2024/06/17/2",
     "https://github.com/python/cpython/commit/01c37f1d0714f5822d34063ca7180b595abf589d",
     "https://github.com/python/cpython/commit/29c97287d205bf2f410f4895ebce3f43b5160524",
     "https://github.com/python/cpython/commit/37324b421b72b7bc9934e27aba85d48d4773002e",
     "https://github.com/python/cpython/commit/542f3272f56f31ed04e74c40635a913fbc12d286",
     "https://github.com/python/cpython/commit/b228655c227b2ca298a8ffac44d14ce3d22f6faa",
     "https://github.com/python/cpython/commit/bce693111bff906ccf9281c22371331aaff766ab",
     "https://github.com/python/cpython/issues/114572",
     "https://github.com/python/cpython/pull/114573",
     "https://mail.python.org/archives/list/security-announce@python.org/thread/BMAK5BCGKYWNJOACVUSLUF6SFGBIM4VP/"
    ],
    "description": "A defect was discovered in the Python “ssl” module where there is a memory\nrace condition with the ssl.SSLContext methods “cert_store_stats()” and\n“get_ca_certs()”. The race condition can be triggered if the methods are\ncalled at the same time as certificates are loaded into the SSLContext,\nsuch as during the TLS handshake with a certificate directory configured.\nThis issue is fixed in CPython 3.10.14, 3.11.9, 3.12.3, and 3.13.0a5.",
    "cvss": [
     {
      "source": "134c704f-9b21-4f2e-91b3-4a467353bcc0",
      "type": "Secondary",
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:N/A:H",
      "metrics": {
       "baseScore": 7.4,
       "exploitabilityScore": 2.2,
       "impactScore": 5.2
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
      ],
      "package": {
       "name": "python",
       "version": "3.9.19"
      }
     },
     "found": {
      "vulnerabilityID": "CVE-2024-0397",
      "versionConstraint": ">= 3.13.0a1, < 3.13.0a5 || >= 3.12.0, < 3.12.3 || >= 3.11.0, < 3.11.9 || < 3.10.14 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "id": "a120648b38c61b3b",
    "name": "python",
    "version": "3.9.19",
    "type": "binary",
    "locations": [
     {
      "path": "/var/lang/bin/python3.9",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/libpython3.9.so.1.0",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.9.19",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2023-36632",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2023-36632",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://docs.python.org/3/library/email.html",
     "https://docs.python.org/3/library/email.utils.html",
     "https://github.com/Daybreak2019/PoC_python3.9_Vul/blob/main/RecursionError-email.utils.parseaddr.py",
     "https://github.com/python/cpython/issues/103800"
    ],
    "description": "The legacy email.utils.parseaddr function in Python through 3.11.4 allows attackers to trigger \"RecursionError: maximum recursion depth exceeded while calling a Python object\" via a crafted argument. This argument is plausibly an untrusted value from an application's input data that was supposed to contain a name and an e-mail address. NOTE: email.utils.parseaddr is categorized as a Legacy API in the documentation of the Python email package. Applications should instead use the email.parser.BytesParser or email.parser.Parser class. NOTE: the vendor's perspective is that this is neither a vulnerability nor a bug. The email package is intended to have size limits and to throw an exception when limits are exceeded; they were exceeded by the example demonstration code.",
    "cvss": [
     {
      "source": "nvd@nist.gov",
      "type": "Primary",
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
      "metrics": {
       "baseScore": 7.5,
       "exploitabilityScore": 3.9,
       "impactScore": 3.6
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
      ],
      "package": {
       "name": "python",
       "version": "3.9.19"
      }
     },
     "found": {
      "vulnerabilityID": "CVE-2023-36632",
      "versionConstraint": "<= 3.11.4 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "id": "a120648b38c61b3b",
    "name": "python",
    "version": "3.9.19",
    "type": "binary",
    "locations": [
     {
      "path": "/var/lang/bin/python3.9",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/libpython3.9.so.1.0",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.9.19",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2024-6923",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-6923",
    "namespace": "nvd:cpe",
    "severity": "Medium",
    "urls": [
     "https://github.com/python/cpython/commit/4766d1200fdf8b6728137aa2927a297e224d5fa7",
     "https://github.com/python/cpython/commit/4aaa4259b5a6e664b7316a4d60bdec7ee0f124d0",
     "https://github.com/python/cpython/issues/121650",
     "https://github.com/python/cpython/pull/122233",
     "https://mail.python.org/archives/list/security-announce@python.org/thread/QH3BUOE2DYQBWP7NAQ7UNHPPOELKISRW/"
    ],
    "description": "There is a MEDIUM severity vulnerability affecting CPython.\n\nThe \nemail module didn’t properly quote newlines for email headers when \nserializing an email message allowing for header injection when an email\n is serialized.",
    "cvss": [
     {
      "source": "134c704f-9b21-4f2e-91b3-4a467353bcc0",
      "type": "Secondary",
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:L/I:L/A:L",
      "metrics": {
       "baseScore": 5.5,
       "exploitabilityScore": 2.1,
       "impactScore": 3.4
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
      ],
      "package": {
       "name": "python",
       "version": "3.9.19"
      }
     },
     "found": {
      "vulnerabilityID": "CVE-2024-6923",
      "versionConstraint": ">= 3.13.0a1, <= 3.13.0rc2 || < 3.12.5 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "id": "a120648b38c61b3b",
    "name": "python",
    "version": "3.9.19",
    "type": "binary",
    "locations": [
     {
      "path": "/var/lang/bin/python3.9",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/libpython3.9.so.1.0",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.9.19",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2023-27043",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2023-27043",
    "namespace": "nvd:cpe",
    "severity": "Medium",
    "urls": [
     "http://python.org",
     "https://github.com/python/cpython/issues/102988",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4ZAEFSFZDNBNJPNOUTLG5COISGQDLMGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/75DTHSTNOFFNAWHXKMDXS7EJWC6W2FUC/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ARI7VDSNTQVXRQFM6IK5GSSLEIYV4VZH/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/BQAKLUJMHFGVBRDPEY57BJGNCE5UUPHW/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/HXYVPEZUA3465AEFX5JVFVP7KIFZMF3N/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N6M5I6OQHJABNEYY555HUMMKX3Y4P25Z/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/NEUNZSZ3CVSM2QWVYH3N2XGOCDWNYUA3/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ORLXS5YTKN65E2Q2NWKXMFS5FWQHRNZW/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/P2MAICLFDDO3QVNHTZ2OCERZQ34R2PIC/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/P2W2BZQIHMCKRI5FNBJERFYMS5PK6TAH/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/PHVGRKQAGANCSGFI3QMYOCIMS4IFOZA5/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/PU6Y2S5CBN5BWCBDAJFTGIBZLK3S2G3J/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QDRDDPDN3VFIYXJIYEABY6USX5EU66AG/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/RDDC2VOX7OQC6OHMYTVD4HLFZIV6PYBC/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/SINP4OVYNB2AGDYI2GS37EMW3H3F7XPZ/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/SOX7BCN6YL7B3RFPEEXPIU5CMTEHJOKR/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/VZXC32CJ7TWDPJO6GY2XIQRO7JZX5FLP/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/XWMBD4LNHWEXRI6YVFWJMTJQUL5WOFTS/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YQVY5C5REXWJIORJIL2FIL3ALOEJEF72/",
     "https://python-security.readthedocs.io/vuln/email-parseaddr-realname.html",
     "https://security.netapp.com/advisory/ntap-20230601-0003/"
    ],
    "description": "The email module of Python through 3.11.3 incorrectly parses e-mail addresses that contain a special character. The wrong portion of an RFC2822 header is identified as the value of the addr-spec. In some applications, an attacker can bypass a protection mechanism in which application access is granted only after verifying receipt of e-mail to a specific domain (e.g., only @company.example.com addresses may be used for signup). This occurs in email/_parseaddr.py in recent versions of Python.",
    "cvss": [
     {
      "source": "nvd@nist.gov",
      "type": "Primary",
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:L/A:N",
      "metrics": {
       "baseScore": 5.3,
       "exploitabilityScore": 3.9,
       "impactScore": 1.4
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
      ],
      "package": {
       "name": "python",
       "version": "3.9.19"
      }
     },
     "found": {
      "vulnerabilityID": "CVE-2023-27043",
      "versionConstraint": "<= 2.7.18 || >= 3.0, <= 3.11 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "id": "a120648b38c61b3b",
    "name": "python",
    "version": "3.9.19",
    "type": "binary",
    "locations": [
     {
      "path": "/var/lang/bin/python3.9",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/libpython3.9.so.1.0",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.9.19",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2024-8088",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-8088",
    "namespace": "nvd:cpe",
    "severity": "Unknown",
    "urls": [
     "https://github.com/python/cpython/commit/795f2597a4be988e2bb19b69ff9958e981cb894e",
     "https://github.com/python/cpython/commit/8c7348939d8a3ecd79d630075f6be1b0c5b41f64",
     "https://github.com/python/cpython/commit/dcc5182f27c1500006a1ef78e10613bb45788dea",
     "https://github.com/python/cpython/commit/e0264a61119d551658d9445af38323ba94fc16db",
     "https://github.com/python/cpython/issues/122905",
     "https://github.com/python/cpython/issues/123270",
     "https://github.com/python/cpython/pull/122906",
     "https://mail.python.org/archives/list/security-announce@python.org/thread/GNFCKVI4TCATKQLALJ5SN4L4CSPSMILU/"
    ],
    "description": "There is a HIGH severity vulnerability affecting the CPython \"zipfile\"\nmodule affecting \"zipfile.Path\". Note that the more common API \"zipfile.ZipFile\" class is unaffected.\n\n\n\n\n\nWhen iterating over names of entries in a zip archive (for example, methods\nof \"zipfile.Path\" like \"namelist()\", \"iterdir()\", etc)\nthe process can be put into an infinite loop with a maliciously crafted\nzip archive. This defect applies when reading only metadata or extracting\nthe contents of the zip archive. Programs that are not handling\nuser-controlled zip archives are not affected.",
    "cvss": [],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
      ],
      "package": {
       "name": "python",
       "version": "3.9.19"
      }
     },
     "found": {
      "vulnerabilityID": "CVE-2024-8088",
      "versionConstraint": "<= 3.13.0 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "id": "a120648b38c61b3b",
    "name": "python",
    "version": "3.9.19",
    "type": "binary",
    "locations": [
     {
      "path": "/var/lang/bin/python3.9",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/libpython3.9.so.1.0",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.9.19",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2024-5642",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-5642",
    "namespace": "nvd:cpe",
    "severity": "Unknown",
    "urls": [
     "http://www.openwall.com/lists/oss-security/2024/06/28/4",
     "https://github.com/python/cpython/commit/39258d3595300bc7b952854c915f63ae2d4b9c3e",
     "https://github.com/python/cpython/issues/121227",
     "https://github.com/python/cpython/pull/23014",
     "https://jbp.io/2024/06/27/cve-2024-5535-openssl-memory-safety.html",
     "https://mail.python.org/archives/list/security-announce@python.org/thread/PLP2JI3PJY33YG6P5BZYSSNU66HASXBQ/",
     "https://security.netapp.com/advisory/ntap-20240726-0005/"
    ],
    "description": "CPython 3.9 and earlier doesn't disallow configuring an empty list (\"[]\") for SSLContext.set_npn_protocols() which is an invalid value for the underlying OpenSSL API. This results in a buffer over-read when NPN is used (see CVE-2024-5535 for OpenSSL). This vulnerability is of low severity due to NPN being not widely used and specifying an empty list likely being uncommon in-practice (typically a protocol name would be configured).",
    "cvss": [],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
      ],
      "package": {
       "name": "python",
       "version": "3.9.19"
      }
     },
     "found": {
      "vulnerabilityID": "CVE-2024-5642",
      "versionConstraint": "< 3.10.0b1 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "id": "a120648b38c61b3b",
    "name": "python",
    "version": "3.9.19",
    "type": "binary",
    "locations": [
     {
      "path": "/var/lang/bin/python3.9",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/libpython3.9.so.1.0",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.9.19",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2024-4032",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-4032",
    "namespace": "nvd:cpe",
    "severity": "Unknown",
    "urls": [
     "http://www.openwall.com/lists/oss-security/2024/06/17/3",
     "https://github.com/python/cpython/commit/22adf29da8d99933ffed8647d3e0726edd16f7f8",
     "https://github.com/python/cpython/commit/40d75c2b7f5c67e254d0a025e0f2e2c7ada7f69f",
     "https://github.com/python/cpython/commit/895f7e2ac23eff4743143beef0f0c5ac71ea27d3",
     "https://github.com/python/cpython/commit/ba431579efdcbaed7a96f2ac4ea0775879a332fb",
     "https://github.com/python/cpython/commit/c62c9e518b784fe44432a3f4fc265fb95b651906",
     "https://github.com/python/cpython/commit/f86b17ac511e68192ba71f27e752321a3252cee3",
     "https://github.com/python/cpython/issues/113171",
     "https://github.com/python/cpython/pull/113179",
     "https://mail.python.org/archives/list/security-announce@python.org/thread/NRUHDUS2IV2USIZM2CVMSFL6SCKU3RZA/",
     "https://security.netapp.com/advisory/ntap-20240726-0004/",
     "https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml",
     "https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml"
    ],
    "description": "The “ipaddress” module contained incorrect information about whether certain IPv4 and IPv6 addresses were designated as “globally reachable” or “private”. This affected the is_private and is_global properties of the ipaddress.IPv4Address, ipaddress.IPv4Network, ipaddress.IPv6Address, and ipaddress.IPv6Network classes, where values wouldn’t be returned in accordance with the latest information from the IANA Special-Purpose Address Registries.\n\nCPython 3.12.4 and 3.13.0a6 contain updated information from these registries and thus have the intended behavior.",
    "cvss": [],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
      ],
      "package": {
       "name": "python",
       "version": "3.9.19"
      }
     },
     "found": {
      "vulnerabilityID": "CVE-2024-4032",
      "versionConstraint": "< 3.12.4 || >= 3.13.0a1, < 3.13.0a6 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "id": "a120648b38c61b3b",
    "name": "python",
    "version": "3.9.19",
    "type": "binary",
    "locations": [
     {
      "path": "/var/lang/bin/python3.9",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/libpython3.9.so.1.0",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.9.19:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.9.19",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "GHSA-r9hx-vwmv-q579",
    "dataSource": "https://github.com/advisories/GHSA-r9hx-vwmv-q579",
    "namespace": "github:language:python",
    "severity": "High",
    "urls": [
     "https://github.com/advisories/GHSA-r9hx-vwmv-q579"
    ],
    "description": "pypa/setuptools vulnerable to Regular Expression Denial of Service (ReDoS)",
    "cvss": [
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
      "metrics": {
       "baseScore": 7.5,
       "exploitabilityScore": 3.9,
       "impactScore": 3.6
      },
      "vendorMetadata": {
       "base_severity": "High",
       "status": "N/A"
      }
     }
    ],
    "fix": {
     "versions": [
      "65.5.1"
     ],
     "state": "fixed"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [
    {
     "id": "CVE-2022-40897",
     "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2022-40897",
     "namespace": "nvd:cpe",
     "severity": "Medium",
     "urls": [
      "https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200",
      "https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be",
      "https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1",
      "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/",
      "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/",
      "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/",
      "https://pyup.io/vulnerabilities/CVE-2022-40897/52495/",
      "https://security.netapp.com/advisory/ntap-20230214-0001/",
      "https://security.netapp.com/advisory/ntap-20240621-0006/"
     ],
     "description": "Python Packaging Authority (PyPA) setuptools before 65.5.1 allows remote attackers to cause a denial of service via HTML in a crafted package or custom PackageIndex page. There is a Regular Expression Denial of Service (ReDoS) in package_index.py.",
     "cvss": [
      {
       "source": "nvd@nist.gov",
       "type": "Primary",
       "version": "3.1",
       "vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
       "metrics": {
        "baseScore": 5.9,
        "exploitabilityScore": 2.2,
        "impactScore": 3.6
       },
       "vendorMetadata": {}
      }
     ]
    }
   ],
   "matchDetails": [
    {
     "type": "exact-direct-match",
     "matcher": "python-matcher",
     "searchedBy": {
      "language": "python",
      "namespace": "github:language:python",
      "package": {
       "name": "setuptools",
       "version": "58.1.0"
      }
     },
     "found": {
      "versionConstraint": "<65.5.1 (python)",
      "vulnerabilityID": "GHSA-r9hx-vwmv-q579"
     }
    }
   ],
   "artifact": {
    "id": "bc6c39be42a59eb7",
    "name": "setuptools",
    "version": "58.1.0",
    "type": "python",
    "locations": [
     {
      "path": "/var/lang/lib/python3.9/site-packages/setuptools-58.1.0.dist-info/METADATA",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/python3.9/site-packages/setuptools-58.1.0.dist-info/RECORD",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/python3.9/site-packages/setuptools-58.1.0.dist-info/top_level.txt",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "python",
    "licenses": [
     "UNKNOWN"
    ],
    "cpes": [
     "cpe:2.3:a:python:setuptools:58.1.0:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:pypi/setuptools@58.1.0",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "GHSA-cx63-2mw6-8hw5",
    "dataSource": "https://github.com/advisories/GHSA-cx63-2mw6-8hw5",
    "namespace": "github:language:python",
    "severity": "High",
    "urls": [
     "https://github.com/advisories/GHSA-cx63-2mw6-8hw5"
    ],
    "description": "setuptools vulnerable to Command Injection via package URL",
    "cvss": [
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 8.8,
       "exploitabilityScore": 2.8,
       "impactScore": 5.9
      },
      "vendorMetadata": {
       "base_severity": "High",
       "status": "N/A"
      }
     }
    ],
    "fix": {
     "versions": [
      "70.0.0"
     ],
     "state": "fixed"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [
    {
     "id": "CVE-2024-6345",
     "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-6345",
     "namespace": "nvd:cpe",
     "severity": "High",
     "urls": [
      "https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0",
      "https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5"
     ],
     "description": "A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.",
     "cvss": [
      {
       "source": "security@huntr.dev",
       "type": "Secondary",
       "version": "3.0",
       "vector": "CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
       "metrics": {
        "baseScore": 8.8,
        "exploitabilityScore": 2.8,
        "impactScore": 5.9
       },
       "vendorMetadata": {}
      }
     ]
    }
   ],
   "matchDetails": [
    {
     "type": "exact-direct-match",
     "matcher": "python-matcher",
     "searchedBy": {
      "language": "python",
      "namespace": "github:language:python",
      "package": {
       "name": "setuptools",
       "version": "58.1.0"
      }
     },
     "found": {
      "versionConstraint": "<70.0.0 (python)",
      "vulnerabilityID": "GHSA-cx63-2mw6-8hw5"
     }
    }
   ],
   "artifact": {
    "id": "bc6c39be42a59eb7",
    "name": "setuptools",
    "version": "58.1.0",
    "type": "python",
    "locations": [
     {
      "path": "/var/lang/lib/python3.9/site-packages/setuptools-58.1.0.dist-info/METADATA",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/python3.9/site-packages/setuptools-58.1.0.dist-info/RECORD",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     },
     {
      "path": "/var/lang/lib/python3.9/site-packages/setuptools-58.1.0.dist-info/top_level.txt",
      "layerID": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0"
     }
    ],
    "language": "python",
    "licenses": [
     "UNKNOWN"
    ],
    "cpes": [
     "cpe:2.3:a:python:setuptools:58.1.0:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:pypi/setuptools@58.1.0",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2024-24791",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-24791",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://go.dev/cl/591255",
     "https://go.dev/issue/67555",
     "https://groups.google.com/g/golang-dev/c/t0rK-qHBqzY/m/6MMoAZkMAgAJ",
     "https://pkg.go.dev/vuln/GO-2024-2963"
    ],
    "description": "The net/http HTTP/1.1 client mishandled the case where a server responds to a request with an \"Expect: 100-continue\" header with a non-informational (200 or higher) status. This mishandling could leave a client connection in an invalid state, where the next request sent on the connection will fail. An attacker sending a request to a net/http/httputil.ReverseProxy proxy can exploit this mishandling to cause a denial of service by sending \"Expect: 100-continue\" requests which elicit a non-informational response from the backend. Each such request leaves the proxy with an invalid connection, and causes one subsequent request using that connection to fail.",
    "cvss": [
     {
      "source": "134c704f-9b21-4f2e-91b3-4a467353bcc0",
      "type": "Secondary",
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
      "metrics": {
       "baseScore": 7.5,
       "exploitabilityScore": 3.9,
       "impactScore": 3.6
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "go-module-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:golang:go:1.22.4:-:*:*:*:*:*:*"
      ],
      "package": {
       "name": "stdlib",
       "version": "go1.22.4"
      }
     },
     "found": {
      "vulnerabilityID": "CVE-2024-24791",
      "versionConstraint": "< 1.21.12 || >= 1.22.0-0, < 1.22.5 (unknown)",
      "cpes": [
       "cpe:2.3:a:golang:go:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "id": "c93538cbf3d8e93b",
    "name": "stdlib",
    "version": "go1.22.4",
    "type": "go-module",
    "locations": [
     {
      "path": "/usr/local/bin/aws-lambda-rie",
      "layerID": "sha256:9a9b5814afc2b76039ada8e0f4445eeec0c487bcf7637b36b20a7f25bb15b369"
     }
    ],
    "language": "go",
    "licenses": [
     "BSD-3-Clause"
    ],
    "cpes": [
     "cpe:2.3:a:golang:go:1.22.4:-:*:*:*:*:*:*"
    ],
    "purl": "pkg:golang/stdlib@1.22.4",
    "upstreams": [],
    "metadataType": "GolangBinMetadata",
    "metadata": {
     "goCompiledVersion": "go1.22.4",
     "architecture": ""
    }
   }
  },
  {
   "vulnerability": {
    "id": "GHSA-34jh-p97f-mpxf",
    "dataSource": "https://github.com/advisories/GHSA-34jh-p97f-mpxf",
    "namespace": "github:language:python",
    "severity": "Medium",
    "urls": [
     "https://github.com/advisories/GHSA-34jh-p97f-mpxf"
    ],
    "description": "urllib3's Proxy-Authorization request header isn't stripped during cross-origin redirects ",
    "cvss": [
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
      "metrics": {
       "baseScore": 4.4,
       "exploitabilityScore": 0.7,
       "impactScore": 3.6
      },
      "vendorMetadata": {
       "base_severity": "Medium",
       "status": "N/A"
      }
     }
    ],
    "fix": {
     "versions": [
      "1.26.19"
     ],
     "state": "fixed"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [
    {
     "id": "CVE-2024-37891",
     "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2024-37891",
     "namespace": "nvd:cpe",
     "severity": "Medium",
     "urls": [
      "https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e",
      "https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf"
     ],
     "description": " urllib3 is a user-friendly HTTP client library for Python. When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected. However, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects. Because this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident. Users should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach. We believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited: 1. Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support. 2. Not disabling HTTP redirects. 3. Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin. Users are advised to update to either version 1.26.19 or version 2.2.2. Users unable to upgrade may use the `Proxy-Authorization` header with urllib3's `ProxyManager`, disable HTTP redirects using `redirects=False` when sending requests, or not user the `Proxy-Authorization` header as mitigations.",
     "cvss": [
      {
       "source": "security-advisories@github.com",
       "type": "Secondary",
       "version": "3.1",
       "vector": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
       "metrics": {
        "baseScore": 4.4,
        "exploitabilityScore": 0.7,
        "impactScore": 3.6
       },
       "vendorMetadata": {}
      }
     ]
    }
   ],
   "matchDetails": [
    {
     "type": "exact-direct-match",
     "matcher": "python-matcher",
     "searchedBy": {
      "language": "python",
      "namespace": "github:language:python",
      "package": {
       "name": "urllib3",
       "version": "1.26.18"
      }
     },
     "found": {
      "versionConstraint": "<1.26.19 (python)",
      "vulnerabilityID": "GHSA-34jh-p97f-mpxf"
     }
    }
   ],
   "artifact": {
    "id": "18a6c87286e142af",
    "name": "urllib3",
    "version": "1.26.18",
    "type": "python",
    "locations": [
     {
      "path": "/var/runtime/urllib3-1.26.18.dist-info/METADATA",
      "layerID": "sha256:734a487fcc033d64cea626bc74e5937eda7014ad9b80d4b3af36f5cce52d2b03"
     },
     {
      "path": "/var/runtime/urllib3-1.26.18.dist-info/RECORD",
      "layerID": "sha256:734a487fcc033d64cea626bc74e5937eda7014ad9b80d4b3af36f5cce52d2b03"
     },
     {
      "path": "/var/runtime/urllib3-1.26.18.dist-info/top_level.txt",
      "layerID": "sha256:734a487fcc033d64cea626bc74e5937eda7014ad9b80d4b3af36f5cce52d2b03"
     }
    ],
    "language": "python",
    "licenses": [
     "MIT"
    ],
    "cpes": [
     "cpe:2.3:a:python:urllib3:1.26.18:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:pypi/urllib3@1.26.18",
    "upstreams": []
   }
  }
 ],
 "source": {
  "type": "image",
  "target": {
   "userInput": "cloudwatch_to_s3:latest",
   "imageID": "sha256:7e36b65761f2da06282720b87b0d5924a7c3478fb655d38e39f8c5eee55b8317",
   "manifestDigest": "sha256:256ebbfc5c837807131c5cdfbfb6b7654b75fdac1783ea0ada96f04027c781af",
   "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
   "tags": [
    "cloudwatch_to_s3:latest"
   ],
   "imageSize": 579981542,
   "layers": [
    {
     "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
     "digest": "sha256:0044b36762beca646f20eb6730e3575ec671a8857b39485f38ba0e715d375483",
     "size": 335259048
    },
    {
     "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
     "digest": "sha256:a8915dcae1e9ecf7d399d9ad10ac6d490a27c0b058bdd7336493ad04e7cedb94",
     "size": 658034
    },
    {
     "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
     "digest": "sha256:91c63ef88078acd534edde8be18b12863689593ea79b6c86671426479a2e0111",
     "size": 397
    },
    {
     "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
     "digest": "sha256:9a9b5814afc2b76039ada8e0f4445eeec0c487bcf7637b36b20a7f25bb15b369",
     "size": 5963928
    },
    {
     "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
     "digest": "sha256:b98baf0908da50e172bd5dedbd8738e0ce574fe1d7229fec407c64e8725685c0",
     "size": 216419370
    },
    {
     "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
     "digest": "sha256:734a487fcc033d64cea626bc74e5937eda7014ad9b80d4b3af36f5cce52d2b03",
     "size": 21663150
    },
    {
     "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
     "digest": "sha256:a9fee2695ab32ff2b4699c3b605f75b64cc4481af914eff473f4037ae53bcd20",
     "size": 8
    },
    {
     "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
     "digest": "sha256:604b2638b886f5a174251db33570afe12d60f3032f10f19ca584c9552b964c25",
     "size": 1219
    },
    {
     "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
     "digest": "sha256:ee32108545f0755778ec959105632baa94507b72d5b381f9739e23c58aa59e60",
     "size": 16388
    }
   ],
   "manifest": "eyJzY2hlbWFWZXJzaW9uIjoyLCJtZWRpYVR5cGUiOiJhcHBsaWNhdGlvbi92bmQuZG9ja2VyLmRpc3RyaWJ1dGlvbi5tYW5pZmVzdC52Mitqc29uIiwiY29uZmlnIjp7Im1lZGlhVHlwZSI6ImFwcGxpY2F0aW9uL3ZuZC5kb2NrZXIuY29udGFpbmVyLmltYWdlLnYxK2pzb24iLCJzaXplIjo0NDUwLCJkaWdlc3QiOiJzaGEyNTY6N2UzNmI2NTc2MWYyZGEwNjI4MjcyMGI4N2IwZDU5MjRhN2MzNDc4ZmI2NTVkMzhlMzlmOGM1ZWVlNTViODMxNyJ9LCJsYXllcnMiOlt7Im1lZGlhVHlwZSI6ImFwcGxpY2F0aW9uL3ZuZC5kb2NrZXIuaW1hZ2Uucm9vdGZzLmRpZmYudGFyLmd6aXAiLCJzaXplIjozNDU5NDgxNjAsImRpZ2VzdCI6InNoYTI1NjowMDQ0YjM2NzYyYmVjYTY0NmYyMGViNjczMGUzNTc1ZWM2NzFhODg1N2IzOTQ4NWYzOGJhMGU3MTVkMzc1NDgzIn0seyJtZWRpYVR5cGUiOiJhcHBsaWNhdGlvbi92bmQuZG9ja2VyLmltYWdlLnJvb3Rmcy5kaWZmLnRhci5nemlwIiwic2l6ZSI6NjYwNDgwLCJkaWdlc3QiOiJzaGEyNTY6YTg5MTVkY2FlMWU5ZWNmN2QzOTlkOWFkMTBhYzZkNDkwYTI3YzBiMDU4YmRkNzMzNjQ5M2FkMDRlN2NlZGI5NCJ9LHsibWVkaWFUeXBlIjoiYXBwbGljYXRpb24vdm5kLmRvY2tlci5pbWFnZS5yb290ZnMuZGlmZi50YXIuZ3ppcCIsInNpemUiOjI1NjAsImRpZ2VzdCI6InNoYTI1Njo5MWM2M2VmODgwNzhhY2Q1MzRlZGRlOGJlMThiMTI4NjM2ODk1OTNlYTc5YjZjODY2NzE0MjY0NzlhMmUwMTExIn0seyJtZWRpYVR5cGUiOiJhcHBsaWNhdGlvbi92bmQuZG9ja2VyLmltYWdlLnJvb3Rmcy5kaWZmLnRhci5nemlwIiwic2l6ZSI6NTk2Nzg3MiwiZGlnZXN0Ijoic2hhMjU2OjlhOWI1ODE0YWZjMmI3NjAzOWFkYThlMGY0NDQ1ZWVlYzBjNDg3YmNmNzYzN2IzNmIyMGE3ZjI1YmIxNWIzNjkifSx7Im1lZGlhVHlwZSI6ImFwcGxpY2F0aW9uL3ZuZC5kb2NrZXIuaW1hZ2Uucm9vdGZzLmRpZmYudGFyLmd6aXAiLCJzaXplIjoyMjMzMzkwMDgsImRpZ2VzdCI6InNoYTI1NjpiOThiYWYwOTA4ZGE1MGUxNzJiZDVkZWRiZDg3MzhlMGNlNTc0ZmUxZDcyMjlmZWM0MDdjNjRlODcyNTY4NWMwIn0seyJtZWRpYVR5cGUiOiJhcHBsaWNhdGlvbi92bmQuZG9ja2VyLmltYWdlLnJvb3Rmcy5kaWZmLnRhci5nemlwIiwic2l6ZSI6MjM5NDkzMTIsImRpZ2VzdCI6InNoYTI1Njo3MzRhNDg3ZmNjMDMzZDY0Y2VhNjI2YmM3NGU1OTM3ZWRhNzAxNGFkOWI4MGQ0YjNhZjM2ZjVjY2U1MmQyYjAzIn0seyJtZWRpYVR5cGUiOiJhcHBsaWNhdGlvbi92bmQuZG9ja2VyLmltYWdlLnJvb3Rmcy5kaWZmLnRhci5nemlwIiwic2l6ZSI6MzA3MiwiZGlnZXN0Ijoic2hhMjU2OmE5ZmVlMjY5NWFiMzJmZjJiNDY5OWMzYjYwNWY3NWI2NGNjNDQ4MWFmOTE0ZWZmNDczZjQwMzdhZTUzYmNkMjAifSx7Im1lZGlhVHlwZSI6ImFwcGxpY2F0aW9uL3ZuZC5kb2NrZXIuaW1hZ2Uucm9vdGZzLmRpZmYudGFyLmd6aXAiLCJzaXplIjo0MDk2LCJkaWdlc3QiOiJzaGEyNTY6NjA0YjI2MzhiODg2ZjVhMTc0MjUxZGIzMzU3MGFmZTEyZDYwZjMwMzJmMTBmMTljYTU4NGM5NTUyYjk2NGMyNSJ9LHsibWVkaWFUeXBlIjoiYXBwbGljYXRpb24vdm5kLmRvY2tlci5pbWFnZS5yb290ZnMuZGlmZi50YXIuZ3ppcCIsInNpemUiOjIwNDgwLCJkaWdlc3QiOiJzaGEyNTY6ZWUzMjEwODU0NWYwNzU1Nzc4ZWM5NTkxMDU2MzJiYWE5NDUwN2I3MmQ1YjM4MWY5NzM5ZTIzYzU4YWE1OWU2MCJ9XX0=",
   "config": "eyJhcmNoaXRlY3R1cmUiOiJhcm02NCIsImNvbmZpZyI6eyJFbnYiOlsiTEFORz1lbl9VUy5VVEYtOCIsIlRaPTovZXRjL2xvY2FsdGltZSIsIlBBVEg9L3Zhci9sYW5nL2JpbjovdXNyL2xvY2FsL2JpbjovdXNyL2Jpbi86L2Jpbjovb3B0L2JpbiIsIkxEX0xJQlJBUllfUEFUSD0vdmFyL2xhbmcvbGliOi9saWI2NDovdXNyL2xpYjY0Oi92YXIvcnVudGltZTovdmFyL3J1bnRpbWUvbGliOi92YXIvdGFzazovdmFyL3Rhc2svbGliOi9vcHQvbGliIiwiTEFNQkRBX1RBU0tfUk9PVD0vdmFyL3Rhc2siLCJMQU1CREFfUlVOVElNRV9ESVI9L3Zhci9ydW50aW1lIiwiQVdTX0NBX0JVTkRMRT0vZXRjL3NzbC9jZXJ0cy9NSVRSRS1jaGFpbi50eHQiXSwiRW50cnlwb2ludCI6WyIvbGFtYmRhLWVudHJ5cG9pbnQuc2giXSwiQ21kIjpbImNsb3Vkd2F0Y2hfdG9fczMubGFtYmRhX2hhbmRsZXIiXSwiV29ya2luZ0RpciI6Ii92YXIvdGFzayIsIkxhYmVscyI6eyJjb20uYW1hem9uYXdzLmxhbWJkYS5wbGF0Zm9ybS5rZXJuZWwiOiJrNTEwZ2EifSwiQXJnc0VzY2FwZWQiOnRydWV9LCJjcmVhdGVkIjoiMjAyNC0wNy0xMVQxMjo1NToyNi4xNDc3NzIzNFoiLCJoaXN0b3J5IjpbeyJjcmVhdGVkIjoiMjAyNC0wNy0xMFQxMTozODoyMloiLCJjcmVhdGVkX2J5IjoiQVJDSElURUNUVVJFIGFybTY0IiwiYXV0aG9yIjoiQVdTIExhbWJkYSIsImVtcHR5X2xheWVyIjp0cnVlfSx7ImNyZWF0ZWQiOiIyMDI0LTA3LTEwVDExOjM4OjIyWiIsImNyZWF0ZWRfYnkiOiJWQVJJQU5UIHY4IiwiYXV0aG9yIjoiQVdTIExhbWJkYSIsImVtcHR5X2xheWVyIjp0cnVlfSx7ImNyZWF0ZWQiOiIyMDI0LTA3LTEwVDExOjM4OjIyWiIsImNyZWF0ZWRfYnkiOiJBREQgZmlsZTo4MmMzOGFhYzljZDcyZjY2YzQ1ZmVjODIxOTM0NmQ3NGFiNDZlZGZmNTM5NzU5ZmFiMDZkNzhiM2ZhOWY0NzExIC8iLCJhdXRob3IiOiJBV1MgTGFtYmRhIn0seyJjcmVhdGVkIjoiMjAyNC0wNy0xMFQxMTozODoyMloiLCJjcmVhdGVkX2J5IjoiQUREIGZpbGU6Njc1NzUzYWFiYjI5Njk4YzhmYTY4ZTU1ZTkzZjUzOGUyZTViOWRmMWE5ODEyZjQ3NjI2M2Q1ZTNhNjIxMTg4NSAvIiwiYXV0aG9yIjoiQVdTIExhbWJkYSJ9LHsiY3JlYXRlZCI6IjIwMjQtMDctMTBUMTE6Mzg6MjJaIiwiY3JlYXRlZF9ieSI6IkFERCBmaWxlOmE2MzM5Y2NmYzA1YmIyMDIzMjM2NzA2OThjNTA0ZTQ2NDRlOGY1NzcyZWNhNjdjODY2ZGNlZmU2Y2Y1NmJkZDQgLyIsImF1dGhvciI6IkFXUyBMYW1iZGEifSx7ImNyZWF0ZWQiOiIyMDI0LTA3LTEwVDExOjM4OjIyWiIsImNyZWF0ZWRfYnkiOiJBREQgZmlsZTo5NzEyMzc4MzI3NGQ0MDJmMTJjMzIwZWE1MWJjYWVkMzgwMzQyOTlmZWJiMzU4ZGVlZmNlM2QwOGYwY2E3NTZkIC8iLCJhdXRob3IiOiJBV1MgTGFtYmRhIn0seyJjcmVhdGVkIjoiMjAyNC0wNy0xMFQxMTozODoyMloiLCJjcmVhdGVkX2J5IjoiQUREIGZpbGU6NzExODkyNzlhYjA4YTFjNmY2ZDMzODYyYjkwODEwNTliZGIzMWRhMDU0MWZhNTI1MmI0Yjg3YTczZGEyZGUwMSAvIiwiYXV0aG9yIjoiQVdTIExhbWJkYSJ9LHsiY3JlYXRlZCI6IjIwMjQtMDctMTBUMTE6Mzg6MjJaIiwiY3JlYXRlZF9ieSI6IkFERCBmaWxlOmQ5NGUwMDEwMTRmODIwNzdiMTYzMDY1OTYwYjkzMjFiODZmYTQ2OWIxZWEzNTIwZDA2Mjc1ZjlmODM4ODM3OGEgLyIsImF1dGhvciI6IkFXUyBMYW1iZGEifSx7ImNyZWF0ZWQiOiIyMDI0LTA3LTEwVDExOjM4OjIyWiIsImNyZWF0ZWRfYnkiOiJXT1JLRElSIC92YXIvdGFzayIsImF1dGhvciI6IkFXUyBMYW1iZGEiLCJlbXB0eV9sYXllciI6dHJ1ZX0seyJjcmVhdGVkIjoiMjAyNC0wNy0xMFQxMTozODoyMloiLCJjcmVhdGVkX2J5IjoiRU5WIExBTkc9ZW5fVVMuVVRGLTgiLCJhdXRob3IiOiJBV1MgTGFtYmRhIiwiZW1wdHlfbGF5ZXIiOnRydWV9LHsiY3JlYXRlZCI6IjIwMjQtMDctMTBUMTE6Mzg6MjJaIiwiY3JlYXRlZF9ieSI6IkVOViBUWj06L2V0Yy9sb2NhbHRpbWUiLCJhdXRob3IiOiJBV1MgTGFtYmRhIiwiZW1wdHlfbGF5ZXIiOnRydWV9LHsiY3JlYXRlZCI6IjIwMjQtMDctMTBUMTE6Mzg6MjJaIiwiY3JlYXRlZF9ieSI6IkVOViBQQVRIPS92YXIvbGFuZy9iaW46L3Vzci9sb2NhbC9iaW46L3Vzci9iaW4vOi9iaW46L29wdC9iaW4iLCJhdXRob3IiOiJBV1MgTGFtYmRhIiwiZW1wdHlfbGF5ZXIiOnRydWV9LHsiY3JlYXRlZCI6IjIwMjQtMDctMTBUMTE6Mzg6MjJaIiwiY3JlYXRlZF9ieSI6IkVOViBMRF9MSUJSQVJZX1BBVEg9L3Zhci9sYW5nL2xpYjovbGliNjQ6L3Vzci9saWI2NDovdmFyL3J1bnRpbWU6L3Zhci9ydW50aW1lL2xpYjovdmFyL3Rhc2s6L3Zhci90YXNrL2xpYjovb3B0L2xpYiIsImF1dGhvciI6IkFXUyBMYW1iZGEiLCJlbXB0eV9sYXllciI6dHJ1ZX0seyJjcmVhdGVkIjoiMjAyNC0wNy0xMFQxMTozODoyMloiLCJjcmVhdGVkX2J5IjoiRU5WIExBTUJEQV9UQVNLX1JPT1Q9L3Zhci90YXNrIiwiYXV0aG9yIjoiQVdTIExhbWJkYSIsImVtcHR5X2xheWVyIjp0cnVlfSx7ImNyZWF0ZWQiOiIyMDI0LTA3LTEwVDExOjM4OjIyWiIsImNyZWF0ZWRfYnkiOiJFTlYgTEFNQkRBX1JVTlRJTUVfRElSPS92YXIvcnVudGltZSIsImF1dGhvciI6IkFXUyBMYW1iZGEiLCJlbXB0eV9sYXllciI6dHJ1ZX0seyJjcmVhdGVkIjoiMjAyNC0wNy0xMFQxMTozODoyMloiLCJjcmVhdGVkX2J5IjoiRU5UUllQT0lOVCBbIFwiL2xhbWJkYS1lbnRyeXBvaW50LnNoXCIgXSIsImF1dGhvciI6IkFXUyBMYW1iZGEiLCJlbXB0eV9sYXllciI6dHJ1ZX0seyJjcmVhdGVkIjoiMjAyNC0wNy0xMFQxMTo0MToyOC45MDE3NTM0NjdaIiwiY3JlYXRlZF9ieSI6Ii9iaW4vc2ggLWMgIyhub3ApICBMQUJFTCBjb20uYW1hem9uYXdzLmxhbWJkYS5wbGF0Zm9ybS5rZXJuZWw9azUxMGdhIiwiZW1wdHlfbGF5ZXIiOnRydWV9LHsiY3JlYXRlZCI6IjIwMjQtMDctMTFUMTI6NTU6MjYuMTMyMTkwMTc0WiIsImNyZWF0ZWRfYnkiOiJDT1BZIGNsb3Vkd2F0Y2hfdG9fczMvcmVxdWlyZW1lbnRzLnR4dCAvdmFyL3Rhc2sgIyBidWlsZGtpdCIsImNvbW1lbnQiOiJidWlsZGtpdC5kb2NrZXJmaWxlLnYwIn0seyJjcmVhdGVkIjoiMjAyNC0wNy0xMVQxMjo1NToyNi4xNDExMjQ3MTVaIiwiY3JlYXRlZF9ieSI6IkNPUFkgY2xvdWR3YXRjaF90b19zMy9jbG91ZHdhdGNoX3RvX3MzLnB5IC92YXIvdGFzayAjIGJ1aWxka2l0IiwiY29tbWVudCI6ImJ1aWxka2l0LmRvY2tlcmZpbGUudjAifSx7ImNyZWF0ZWQiOiIyMDI0LTA3LTExVDEyOjU1OjI2LjE0Nzc3MjM0WiIsImNyZWF0ZWRfYnkiOiJDT1BZIC4vTUlUUkUtY2hhaW4udHh0IC9ldGMvc3NsL2NlcnRzL01JVFJFLWNoYWluLnR4dCAjIGJ1aWxka2l0IiwiY29tbWVudCI6ImJ1aWxka2l0LmRvY2tlcmZpbGUudjAifSx7ImNyZWF0ZWQiOiIyMDI0LTA3LTExVDEyOjU1OjI2LjE0Nzc3MjM0WiIsImNyZWF0ZWRfYnkiOiJFTlYgQVdTX0NBX0JVTkRMRT0vZXRjL3NzbC9jZXJ0cy9NSVRSRS1jaGFpbi50eHQiLCJjb21tZW50IjoiYnVpbGRraXQuZG9ja2VyZmlsZS52MCIsImVtcHR5X2xheWVyIjp0cnVlfSx7ImNyZWF0ZWQiOiIyMDI0LTA3LTExVDEyOjU1OjI2LjE0Nzc3MjM0WiIsImNyZWF0ZWRfYnkiOiJDTUQgW1wiY2xvdWR3YXRjaF90b19zMy5sYW1iZGFfaGFuZGxlclwiXSIsImNvbW1lbnQiOiJidWlsZGtpdC5kb2NrZXJmaWxlLnYwIiwiZW1wdHlfbGF5ZXIiOnRydWV9XSwib3MiOiJsaW51eCIsInJvb3RmcyI6eyJ0eXBlIjoibGF5ZXJzIiwiZGlmZl9pZHMiOlsic2hhMjU2OjAwNDRiMzY3NjJiZWNhNjQ2ZjIwZWI2NzMwZTM1NzVlYzY3MWE4ODU3YjM5NDg1ZjM4YmEwZTcxNWQzNzU0ODMiLCJzaGEyNTY6YTg5MTVkY2FlMWU5ZWNmN2QzOTlkOWFkMTBhYzZkNDkwYTI3YzBiMDU4YmRkNzMzNjQ5M2FkMDRlN2NlZGI5NCIsInNoYTI1Njo5MWM2M2VmODgwNzhhY2Q1MzRlZGRlOGJlMThiMTI4NjM2ODk1OTNlYTc5YjZjODY2NzE0MjY0NzlhMmUwMTExIiwic2hhMjU2OjlhOWI1ODE0YWZjMmI3NjAzOWFkYThlMGY0NDQ1ZWVlYzBjNDg3YmNmNzYzN2IzNmIyMGE3ZjI1YmIxNWIzNjkiLCJzaGEyNTY6Yjk4YmFmMDkwOGRhNTBlMTcyYmQ1ZGVkYmQ4NzM4ZTBjZTU3NGZlMWQ3MjI5ZmVjNDA3YzY0ZTg3MjU2ODVjMCIsInNoYTI1Njo3MzRhNDg3ZmNjMDMzZDY0Y2VhNjI2YmM3NGU1OTM3ZWRhNzAxNGFkOWI4MGQ0YjNhZjM2ZjVjY2U1MmQyYjAzIiwic2hhMjU2OmE5ZmVlMjY5NWFiMzJmZjJiNDY5OWMzYjYwNWY3NWI2NGNjNDQ4MWFmOTE0ZWZmNDczZjQwMzdhZTUzYmNkMjAiLCJzaGEyNTY6NjA0YjI2MzhiODg2ZjVhMTc0MjUxZGIzMzU3MGFmZTEyZDYwZjMwMzJmMTBmMTljYTU4NGM5NTUyYjk2NGMyNSIsInNoYTI1NjplZTMyMTA4NTQ1ZjA3NTU3NzhlYzk1OTEwNTYzMmJhYTk0NTA3YjcyZDViMzgxZjk3MzllMjNjNThhYTU5ZTYwIl19LCJ2YXJpYW50IjoidjgifQ==",
   "repoDigests": [],
   "architecture": "arm64",
   "os": "linux",
   "labels": {
    "com.amazonaws.lambda.platform.kernel": "k510ga"
   }
  }
 },
 "distro": {
  "name": "amazonlinux",
  "version": "2",
  "idLike": [
   "centos",
   "rhel",
   "fedora"
  ]
 },
 "descriptor": {
  "name": "grype",
  "version": "0.79.3",
  "configuration": {
   "output": [
    "json"
   ],
   "file": "",
   "distro": "",
   "add-cpes-if-none": false,
   "output-template-file": "",
   "check-for-app-update": true,
   "only-fixed": false,
   "only-notfixed": false,
   "ignore-wontfix": "",
   "platform": "",
   "search": {
    "scope": "squashed",
    "unindexed-archives": false,
    "indexed-archives": true
   },
   "ignore": [
    {
     "vulnerability": "",
     "reason": "",
     "namespace": "",
     "fix-state": "",
     "package": {
      "name": "kernel-headers",
      "version": "",
      "language": "",
      "type": "rpm",
      "location": "",
      "upstream-name": "kernel"
     },
     "vex-status": "",
     "vex-justification": "",
     "match-type": "exact-indirect-match"
    },
    {
     "vulnerability": "",
     "reason": "",
     "namespace": "",
     "fix-state": "",
     "package": {
      "name": "linux-headers-.*",
      "version": "",
      "language": "",
      "type": "deb",
      "location": "",
      "upstream-name": "linux"
     },
     "vex-status": "",
     "vex-justification": "",
     "match-type": "exact-indirect-match"
    },
    {
     "vulnerability": "",
     "reason": "",
     "namespace": "",
     "fix-state": "",
     "package": {
      "name": "linux-libc-dev",
      "version": "",
      "language": "",
      "type": "deb",
      "location": "",
      "upstream-name": "linux"
     },
     "vex-status": "",
     "vex-justification": "",
     "match-type": "exact-indirect-match"
    }
   ],
   "exclude": [],
   "db": {
    "cache-dir": "/Users/atang/Library/Caches/grype/db",
    "update-url": "https://toolbox-data.anchore.io/grype/databases/listing.json",
    "ca-cert": "",
    "auto-update": true,
    "validate-by-hash-on-start": false,
    "validate-age": true,
    "max-allowed-built-age": 432000000000000,
    "update-available-timeout": 30000000000,
    "update-download-timeout": 120000000000
   },
   "externalSources": {
    "enable": false,
    "maven": {
     "searchUpstreamBySha1": true,
     "baseUrl": "https://search.maven.org/solrsearch/select"
    }
   },
   "match": {
    "java": {
     "using-cpes": false
    },
    "dotnet": {
     "using-cpes": false
    },
    "golang": {
     "using-cpes": false,
     "always-use-cpe-for-stdlib": true,
     "allow-main-module-pseudo-version-comparison": false
    },
    "javascript": {
     "using-cpes": false
    },
    "python": {
     "using-cpes": false
    },
    "ruby": {
     "using-cpes": false
    },
    "rust": {
     "using-cpes": false
    },
    "stock": {
     "using-cpes": true
    }
   },
   "fail-on-severity": "",
   "registry": {
    "insecure-skip-tls-verify": false,
    "insecure-use-http": false,
    "auth": null,
    "ca-cert": ""
   },
   "show-suppressed": false,
   "by-cve": false,
   "name": "",
   "default-image-pull-source": "",
   "vex-documents": [],
   "vex-add": [],
   "match-upstream-kernel-headers": false
  },
  "db": {
   "built": "2024-08-29T01:31:52Z",
   "schemaVersion": 5,
   "location": "/Users/atang/Library/Caches/grype/db/5",
   "checksum": "sha256:755b4c6811bb129d58b2f2a7ff01473c8cbcd08f1916cc6a250144601a7d1d39",
   "error": null
  },
  "timestamp": "2024-08-29T13:47:41.623667-04:00"
 }
}
